packages/api-electron/src/preload/window.ts

```diff
@@ -28,6 +29,19 @@ class Window implements ssf.Window {
       return this;
     }
 
+    if (!isUrlPattern.test(options.url) && options.url !== 'about:blank') {
```

Comment:
Can you add a comment above this block of code to explain why you are manipulating the URL?

Also, the code mutates the `option` object that the user passes in (which is a bit nasty), I think it would be better to move this code into a separate function that returns a new URL string.